eeg_flow/tasks/epochs_evoked.py

```diff
@@ -0,0 +1,434 @@
+
+import numpy as np
+import pandas as pd
+from autoreject import get_rejection_threshold
+from mne import Epochs, find_events
+from mne.epochs import make_metadata
+from mne.io import read_raw_fif
+from scipy.stats import norm
+
+from ..config import load_config
+from ..utils._docs import fill_doc
+from ..utils.bids import get_fname, get_derivative_folder
+from ..utils.concurrency import lock_files
+
+
+@fill_doc
+def behav_prep_epoching(
+    participant: str,
+    group: str,
+    task: str,
+    run: int,
+    *,
+    timeout: float = 10,
+) -> None:
+    """Compute behav report and output metadata.
+    Parameters
+    ----------
+    %(participant)s
+    %(group)s
+    %(task)s
+    %(run)s
+    """
+    # prepare folders
+    _, derivatives_folder_root, _ = load_config()
+    fname_stem = get_fname(participant, group, task, run)
+    derivatives_folder = get_derivative_folder(
+        derivatives_folder_root, participant, group, task, run
+    )
+
+    # lock the output derivative files
+    # create locks
+    derivatives = [
+        derivatives_folder / (fname_stem + "_step9_a-metadata.csv"),
+        derivatives_folder / (fname_stem + "_step9_b-behav.txt"),
+        derivatives_folder / (fname_stem + "_step9_c1-cleaned-epo.fif"),
+        derivatives_folder / (fname_stem + "_step9_c2-drop-epochs.csv"),
+        derivatives_folder / (fname_stem + "_step9_c3-drop-channel-log.csv"),
+        derivatives_folder / (fname_stem + "_step9_d1-standard_evoked-ave.fif"),
+        derivatives_folder / (fname_stem + "_step9_d2-target_evoked-ave.fif"),
+        derivatives_folder / (fname_stem + "_step9_d3-novel_evoked-ave.fif"),
+    ]
+
+    locks = lock_files(*derivatives, timeout=timeout)
+
+    try:
+        _behav_prep_epoching(participant, group, task, run)
+    finally:
+        for lock in locks:
+            lock.release()
+        del locks
+    return
+
+
+@fill_doc
+def _behav_prep_epoching(
+    participant: str,
+    group: str,
+    task: str,
+    run: int,
+) -> None:
+    """Compute behav report and output metadata.
+    Parameters
+    ----------
+    %(participant)s
+    %(group)s
+    %(task)s
+    %(run)s
+    """
+    # prepare folders
+    _, derivatives_folder_root, _ = load_config()
+    fname_stem = get_fname(participant, group, task, run)
+    derivatives_folder = get_derivative_folder(
+        derivatives_folder_root, participant, group, task, run
+    )
+
+    # load previous steps
+    # # load raw_fit recording
+    raw = read_raw_fif(
+        derivatives_folder / (fname_stem + "_step7_preprocessed_raw.fif"),
+        preload=True,
+    )
+
+    events = find_events(raw, stim_channel="TRIGGER")
+
+    events_id = dict(standard=1, target=2, novel=3, response=64)
+    row_events = ["standard", "target", "novel"]
+
+    metadata, events, event_id = create_metadata(events, events_id, raw, row_events)
+    FNAME_METADATA = derivatives_folder / (fname_stem + "_step9_a-metadata.csv")
+    metadata.to_csv(FNAME_METADATA)
```

Comment:
Variables defined in capital letters are variables that the user must explicitly set or constants. For instance:

```
if __name__ == `__main__`:
    VAR = 3
    foo(VAR)
```

By definition, they can not be present inside a function, they always live in the `main` namespace. But you can find them at the top of a module, as a constant. For instance: https://github.com/mne-tools/mne-lsl/blob/6b274b8a9a2efd04990a5fa83578cd97eba9f034/mne_lsl/utils/meas_info.py#L27-L45

See also: https://realpython.com/python-pep8/#naming-styles